tests/models/cpmant/test_modeling_cpmant.py

```diff
@@ -0,0 +1,234 @@
+        hidden_act="gelu",
+        hidden_dropout_prob=0.1,
+        attention_dropout_prob=0.1,
+        max_position_embeddings=512,
+        type_vocab_size=16,
+        type_sequence_label_size=2,
+        initializer_range=0.02,
+        num_labels=3,
+        num_choices=4,
+        scope=None,
+    ):
+        self.parent = parent
+        self.batch_size = batch_size
+        self.seq_length = seq_length
+        self.is_training = is_training
+        self.use_token_type_ids = use_token_type_ids
+        self.use_input_mask = use_input_mask
+        self.use_labels = use_labels
+        self.use_mc_token_ids = use_mc_token_ids
+        self.vocab_size = vocab_size
+        self.hidden_size = hidden_size
+        self.num_hidden_layers = num_hidden_layers
+        self.num_attention_heads = num_attention_heads
+        self.intermediate_size = intermediate_size
+        self.hidden_act = hidden_act
+        self.hidden_dropout_prob = hidden_dropout_prob
+        self.attention_dropout_prob = attention_dropout_prob
+        self.max_position_embeddings = max_position_embeddings
+        self.type_vocab_size = type_vocab_size
+        self.type_sequence_label_size = type_sequence_label_size
+        self.initializer_range = initializer_range
+        self.num_labels = num_labels
+        self.num_choices = num_choices
+        self.scope = None
+        self.bos_token_id = vocab_size - 1
+        self.eos_token_id = vocab_size - 1
+        self.pad_token_id = vocab_size - 1
+
+    def prepare_config_and_inputs(self):
+        input_ids = {}
+        input_ids["input_ids"] = ids_tensor([self.batch_size, self.seq_length], self.vocab_size).type(torch.int32)
+
+        config = self.get_config()
+
+        return (config, input_ids)
+
+    def get_config(self):
+        return CPMAntConfig.from_pretrained("openbmb/cpm-ant-10b")
+
+    def create_and_check_cpmant_model(self, config, input_ids, *args):
+        model = CPMAntModel(config=config)
+        model.to(torch_device)
+        model.eval()
+
+        hidden_states = model(**input_ids).last_hidden_state
+
+        self.parent.assertEqual(hidden_states.shape, (self.batch_size, self.seq_length, config.dim_model))
+
+    def create_and_check_lm_head_model(self, config, input_ids, *args):
+        model = CPMAntForCausalLM(config)
+        model.to(torch_device)
+        for k, v in input_ids.items():
+            input_ids[k] = v.to(torch_device)
+        model.eval()
+
+        model_output = model(**input_ids)
+        self.parent.assertEqual(
+            model_output.logits.shape,
+            (self.batch_size, self.seq_length, config.vocab_size + config.prompt_types * config.prompt_length),
+        )
+
+    def prepare_config_and_inputs_for_common(self):
+        config, input_ids = self.prepare_config_and_inputs()
+        inputs_dict = {"input_ids": input_ids}
+        return config, inputs_dict
+
+
+@require_torch
+class CPMAntModelTest(unittest.TestCase):
```

Comment:
You need this class to be inherited from `ModelTesterMixin`, otherwise the common tests won't be run
You need to import it from: `from ...test_modeling_common import ModelTesterMixin`
Also, you need a separate tester for `xxxForCausalLM` class as the tester needs to be inherited from `GenerationTesterMixin`
Check [here](https://github.com/huggingface/transformers/blob/b29e2dcaff114762e65eaea739ba1076fc5d1c84/tests/models/gptsan_japanese/test_modeling_gptsan_japanese.py#L162) for more examples.

Reply:
solved